front/src/components/Dialogs/SearchDialog.vue

```diff
@@ -0,0 +1,165 @@
+<template>
+  <q-dialog ref="dialogRef" @hide="onDialogHide">
+    <q-card style="min-width: 80%">
+      <q-card-section class="text-h6"> Search for CTF/Task </q-card-section>
+      <q-separator />
+      <q-card-section>
+        <q-input
+          v-model="searchText"
+          filled
+          label="What are you searching for?"
+          hint="Search for title or description"
+          autofocus
+          :loading="loading"
+          @update:model-value="onSearchChange"
+          @keypress.enter="submit"
+        />
+      </q-card-section>
+
+      <q-card-section v-if="!!items.length">
+        <q-list bordered separator>
+          <q-item
+            v-for="(item, i) in items"
+            :key="item.id"
+            :focused="i === selectedItemIndex"
+            clickable
+            @click="onItemSelected(item)"
+          >
+            <q-item-section>
+              <span>
+                {{ !!item.ctf ? item.ctf.title + ' / ' : '' }}
+                <b>{{ item.title }}</b>
+              </span>
+            </q-item-section>
+
+            <q-item-section side top>
+              <q-badge color="teal" :label="item.__typename" />
+            </q-item-section>
+          </q-item>
+        </q-list>
+      </q-card-section>
+      <q-card-section>
+        <span>This search bar can also be opened by using CTRL+K</span>
+      </q-card-section>
+    </q-card>
+  </q-dialog>
+</template>
+
+<script lang="ts">
+import hotkeys from 'hotkeys-js';
+import { useDialogPluginComponent } from 'quasar';
+import ctfnote from 'src/ctfnote';
+import { safeSlugify } from 'src/ctfnote/ctfs';
+import { Ctf, Task } from 'src/ctfnote/models';
+import { defineComponent, onMounted, Ref, ref } from 'vue';
+
+export default defineComponent({
+  emits: useDialogPluginComponent.emits,
+  setup() {
+    const { dialogRef, onDialogHide, onDialogOK, onDialogCancel } =
+      useDialogPluginComponent();
+
+    const items: Ref<Array<Ctf | Task>> = ref([]);
+    const selectedItemIndex = ref(0);
+    const searchText = ref('');
+
+    const updateSelectedIndex = () => {
+      if (selectedItemIndex.value >= items.value.length)
+        selectedItemIndex.value = items.value.length - 1;
+
+      if (selectedItemIndex.value < 0) selectedItemIndex.value = 0;
+    };
+
+    const previousShortcut = 'command+p, ctrl+p, up';
+    const nextShortcut = 'command+n, ctrl+n, down';
+
+    onMounted(() => {
+      hotkeys.filter = function () {
+        return true;
+      };
+
+      hotkeys(previousShortcut, (event) => {
+        event.stopImmediatePropagation();
+        event.preventDefault();
+        selectedItemIndex.value += -1;
+        updateSelectedIndex();
+      });
+
+      hotkeys(nextShortcut, (event) => {
+        event.stopImmediatePropagation();
+        event.preventDefault();
+        selectedItemIndex.value += 1;
+        updateSelectedIndex();
+      });
+    });
```

Comment:
When I close the dialog, the shortcuts are not removed. Therefore, I can not use the arrow-buttons anymore when importing tasks with the raw parser.
However, this is easy to fix:
```diff
diff --git a/front/src/components/Dialogs/SearchDialog.vue b/front/src/components/Dialogs/SearchDialog.vue
index aa977ae9..2507b1c5 100644
--- a/front/src/components/Dialogs/SearchDialog.vue
+++ b/front/src/components/Dialogs/SearchDialog.vue
@@ -51,7 +51,7 @@ import { useDialogPluginComponent } from 'quasar';
 import ctfnote from 'src/ctfnote';
 import { safeSlugify } from 'src/ctfnote/ctfs';
 import { Ctf, Task } from 'src/ctfnote/models';
-import { defineComponent, onMounted, Ref, ref } from 'vue';
+import { defineComponent, onMounted, onUnmounted, Ref, ref } from 'vue';
 
 export default defineComponent({
   emits: useDialogPluginComponent.emits,
@@ -93,6 +93,11 @@ export default defineComponent({
       });
     });
 
+    onUnmounted(() => {
+      hotkeys.unbind(previousShortcut);
+      hotkeys.unbind(nextShortcut);
+    });
+
     return {
       dialogRef,
       loading: false,
```

Reply:
Nice catch ! pushing now